lib/danger/ci_source/teamcity.rb

```diff
@@ -0,0 +1,19 @@
+module Danger
+  module CISource
+    class TeamCity < CI
+      def self.validates?(env)
+        env.has_key? 'TEAMCITY_VERSION'
+      end
+
+      def supported_request_sources
+        @supported_request_sources ||= [Danger::RequestSources::GitHub]
+      end
+
+      def initialize(env)
+        self.repo_slug       = env['GITHUB_REPO_SLUG']
+        self.pull_request_id = env['GITHUB_PULL_REQUEST_ID'].to_i
+        self.repo_url        = env['GITHUB_REPO_URL']
```

Comment:
Unfortunately, TeamCity doesn't export GitHub specific info via environment variables. So you have to add these environment variables manually to your project or build configuration.